File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.scala
import scala.collection.mutable
object Solution {
    def lengthOfLongestSubstring(s: String): Int = {
        val view = mutable.HashSet[Char]()
        var left = 0
        var res = 0

        for (right <- s.indices) {
            while(view.contains(s(right))){
                view.remove(s(left))
                left+=1
            } 
            view.add(s(right))
            res = math.max(res,right-left+1)  
        }
        res        
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        view = set()
        left = 0
        res = 0

        for right in range(len(s)):
            while s[right] in view:
                view.remove(s[left])
                left +=1

            view.add(s[right])
            res = max(res, right-left+1)
        return res
        